flink-table/flink-table-storage/src/main/java/org/apache/flink/table/storage/filestore/lsm/sst/SstFileMeta.java

```diff
@@ -130,7 +130,9 @@ public int hashCode() {
                 rowCount,
                 minKey,
                 maxKey,
-                stats,
+                // by default, hash code of arrays are computed by reference, not by content.
+                // so we must use Arrays.hashCode to hash by content.
+                Arrays.hashCode(stats),
```

Comment:
Can you add test for this?

Reply:
This is already used by `IntervalPartitionTest`. Without this, the comparison will fail.